File: code_examples/3-Structured_Problem_Solving/11-Compoud_Boolean/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module uses a compound boolean statement
"""


def main() -> None:
    """The main() this function uses a compound boolean statement, returns None."""


    # input
    term_mark = int(input("Enter term mark (as %): "))
    project_mark = int(input("Enter project mark (as %): "))
    print("")

    # process & output
    if term_mark >= 50 and project_mark >= 50:
        print("You passed the course.")
    else:
        print("You did not pass the course.")

    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/3-Structured_Problem_Solving/11-Compoud_Boolean/C/main.c
// Copyright (c) 2020 Mr. Coxall All rights reserved.
//
// Created by: Mr. Coxall
// Created on: Sep 2020
// This program checks for a runtime error

#include <stdio.h>

int main() {
    // this function checks for a runtime error
    int integerValue = 0;
    int scanErrorCode = 0;

    // input
    printf("Enter an integer: ");
    scanErrorCode = scanf("%d", &integerValue);

    // process and output
    if (scanErrorCode == 1) {
        printf("You entered: %d\n", integerValue);
    } else {
        printf("You did not enter an integer.\n");
    }
    printf("Thanks for playing.\n");

        printf("\nDone.\n");
    return 0;
}
